### DISCUSSION TITLE
AG Grid with Conditional Cell Formatting

### DISCUSSION THREAD

**Discussion by johancj** [zauberzeug/nicegui#1103] [Q&A] [answered]
### Question

Hi, 

I found the following example which is close to what I'm trying to achieve, but not quite: https://github.com/zauberzeug/nicegui/discussions/754.

I would like conditional cell formatting where the color should be green for a cell if the cell is equal to the value of another column in the same row. 

How can this be done? 

Best regards
Johan Christian Jenssen

**Comment by falkoschindler** [comment, marked as answer]
If I understand correctly, you're looking for something like this:

```py
ui.add_head_html('''
<style>
.cell-fail { background-color: #f44336; }
.cell-pass { background-color: #4caf50; }
</style>
''')

grid = ui.aggrid({
    'columnDefs': [
        {'headerName': 'Name', 'field': 'name'},
        {'headerName': 'Age', 'field': 'age'},
        {'headerName': 'Candles', 'field': 'candles'},
    ],
    'rowData': [
        {'name': 'Alice', 'age': 18, 'candles': 18},
        {'name': 'Bob', 'age': 21, 'candles': 21},
        {'name': 'Carol', 'age': 42, 'candles': 43},
    ],
})

async def format() -> None:
    await ui.run_javascript(f'''
        getElement({grid.id}).gridOptions.columnApi.getColumn("candles").getColDef().cellClassRules = {{
            "cell-fail": x => x.value != x.data.age,
            "cell-pass": x => x.value == x.data.age,
        }};
        getElement({grid.id}).gridOptions.api.refreshCells();
    ''', respond=False)

app.on_connect(format)
```

With a condition like `x => x.value != x.data.age` you can compare the current cell value `x.value` with another column `x.data.age`.

**Reply by johancj** [reply]
Thank you so much! This was exactly what I was looking for! 😃
